nomad/state/schema.go

```diff
@@ -611,10 +618,22 @@ func schedulerConfigTableSchema() *memdb.TableSchema {
 				Name:         "id",
 				AllowMissing: true,
 				Unique:       true,
-				// This indexer ensures that this table is a singleton
-				Indexer: &memdb.ConditionalIndex{
-					Conditional: func(obj interface{}) (bool, error) { return true, nil },
-				},
+				Indexer:      singletonRecord, // we store only 1 scheduler config
+			},
+		},
+	}
+}
+
+// clusterMetaTableSchema returns the MemDB schema for the scheduler config table.
+func clusterMetaTableSchema() *memdb.TableSchema {
+	return &memdb.TableSchema{
+		Name: "cluster_meta",
+		Indexes: map[string]*memdb.IndexSchema{
+			"id": {
+				Name:         "id",
+				AllowMissing: false,
```

Comment:
Hm, why is AllowMissing `false` here and `true` for `scheduler_config` above? I honestly don't know what the behavior this flag controls based on the docs: https://pkg.go.dev/github.com/hashicorp/go-memdb?tab=doc#IndexSchema

Reply:
My understanding from https://github.com/hashicorp/go-memdb/blob/v1.0.4/txn.go#L256
is that it's analogous to something like MySQL's `NOT NULL`. We shouldn't be inserting an empty ClusterID, so it seems correct to set `AllowMissing: false` here. I dunno about `scheduler_config`

Reply:
Cool! Thanks for tracking down that code. The field comment in memdb is a little oblique.